### PR TITLE
fix a problem when we try to create image with custom repo

### PR DESCRIPTION
**Pull request by mgold1234**
related to THEEDGE-2405
the problem was that we need to copy the orgID to customRepo table

Signed-off-by: mgold1234 <mgold@redhat.com>

# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

What is it?

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation update
- [ ] Tests update
- [ ] Refactor

# Checklist:

- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] I run `make pre-commit` to check fmt/vet/lint/test-no-fdo

### CODE REVIEW

**Review by loadtheaccumulator — DISMISSED** [review]
Have a question on how the repos are being reconstructed and what is passed back.

File: pkg/services/images.go [hunk 1]
```diff
@@ -82,29 +82,27 @@ type ImageService struct {
...
+			if result.Error == gorm.ErrRecordNotFound {
+				return nil, new(ThirdPartyRepositoryNotFound)
+			}
+			return nil, result.Error
+		}
+		imagesRepos = append(imagesRepos, repo)
 	}
-
-	return nil
+	return &imagesRepos, nil
```
[the remark this answers is not in the document]
Comment (ldjebran) [reply pkg/services/images.go:105]: I think if any of the repos , is not from org_id we will return new(ThirdPartyRepositoryNotFound) as we are using filter db.Org(orgID, "")  
if all the repos are from org_id , we will reach line 105

[the remark this answers is not in the document]
Comment (loadtheaccumulator) [reply pkg/services/images.go:105]: That's the way I read it. But why recreate image.ThirdPartyRepositories if what goes in will equal what comes out?

[the remark this answers is not in the document]
Comment (ldjebran) [reply pkg/services/images.go:105]: we are getting the repos from the ui without orgIDS, if we do not get the original ones from db 
gorm will fail when saving , as we have a BefroreCreate that check's orgID if empty
gorm will try a create but fail on the BeforeCreate hook.

it appears , that gorm is always trying to create the extended model of a model. and that's what's  happening to repos,

**Review by loadtheaccumulator — COMMENTED** [review]
Got the answer I needed. Thanks!

File: pkg/services/images.go [hunk 2]
```diff
@@ -82,29 +82,27 @@ type ImageService struct {
 	RepoService  RepoServiceInterface
 }
 
-// ValidateAllImageReposAreFromOrgID validates the account for Third Party Repositories
-func ValidateAllImageReposAreFromOrgID(orgID string, repos []models.ThirdPartyRepo) error {
+// ValidateAllImageReposAreFromOrgID validates the OrgID for Third Party Repositories
+func ValidateAllImageReposAreFromOrgID(orgID string, repos []models.ThirdPartyRepo) (*[]models.ThirdPartyRepo, error) {
```
Comment (ldjebran) [inline pkg/services/images.go:86]: May be we should rename this function as it's not only validate, what do you think ?

**Review by ldjebran — COMMENTED** [review]
Left one question

File: pkg/services/images_test.go [hunk 3]
```diff
@@ -541,7 +541,7 @@ var _ = Describe("Image Service Test", func() {
 				repo2 := models.ThirdPartyRepo{OrgID: orgID, Name: faker.UUIDHyphenated(), URL: "https://repo2.simple.com"}
 				result = db.DB.Create(&repo2)
 				Expect(result.Error).ToNot(HaveOccurred())
-				err := services.ValidateAllImageReposAreFromOrgID(orgID, []models.ThirdPartyRepo{repo1, repo2})
+				_, err := services.ValidateAllImageReposAreFromOrgID(orgID, []models.ThirdPartyRepo{repo1, repo2})
```
Comment (ldjebran) [inline pkg/services/images_test.go:544]: maybe check that the len(returned repos) ==  2
and that the returned repos correspond to the supplied one

Comment (ldjebran) [inline pkg/services/images_test.go:544]: you may supply only repo ids here to simulate the worst case 
may be like 
```go
...  := services.ValidateAllImageReposAreFromOrgID(orgID, []models.ThirdPartyRepo{{ID: repo1.ID}, {ID: repo2.ID}})

```

File: pkg/services/images.go (hunk 2 above)
Comment (mgold1234) [reply pkg/services/images.go:86]: WDYT about ths name - GetImageReposbyOrgID

**Review by ldjebran — APPROVED** [review]

**Pull request merged**